modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/DartDioNextClientCodegen.java

```diff
@@ -278,78 +279,217 @@ public void postProcessModelProperty(CodegenModel model, CodegenProperty propert
         }
     }
 
+    private void fixEnumParam(String opName,IJsonSchemaValidationProperties param) {
```

Comment:
any of these 'fix' methods need some nice documentation in my opinion. A simple description of what this method does and what if fixes.

Reply:
will do, but these are to compensate for incorrect behavior in the core modules

Reply:
ok, definitely in need of some good documentation then! Would be good to explain what they compensate for and at what point that could be removed.

Reply:
I added some comments, let me know if more are needed